MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -110,29 +110,27 @@ class NavigationServiceTests: XCTestCase {
         }
     }
 
-    func testAdvancingToFutureStepAndNotRerouting() {
+    func testNotReroutingForAllSteps() {
```

Comment:
I see that when using nav. native `9.0.4` this test was working non deterministically. E.g. here check for advancing to future step is made only for steps 1 and 3 (why not for all?). When I check steps advancing for all steps I see that `stepIndex` is not increased for steps other than 1 and 3 (e.g. we remain on stepIndex 1 (instead of 2) for location with index 1 and on stepIndex 3 for locations 4 and 5). Explanation for this is that stepIndex is tied to value stored in nav. native class `MBNavigationStatus`. 
I've modified test to check whether locations in list are on route, I think there is no reliable way to check in what case `stepIndex` will be increased.